### DISCUSSION TITLE
Question: How alive is the project? (Does contributing make sense?)

### DISCUSSION THREAD

**Discussion by ZainlessBrombie** [iceiix/stevenarella#436] [General]
Heyho!

as a plugin developer, server owner and general not-fond-of-microsoft rustacean, I was very annoyed with Microsofts move to make their Account mandatory.

Now I know the readme says "don't expect this to go anywhere", but I'd be willing to _try_ to get this project to a playable state (Mobs, graphics, quality of life, chat) if you say it is worth it and you think MS won't come swooping in with their lawyers.  
Edit: None of this would be commercial use so I guess it is okay, right?

There would be a few things I would need / want to add, apart from MC features though:  
1) Optionally using no account. Meaning it can't use the default texture pack by itself, unless locally available.  
2) Optionaly using a "stevenarella" - account. This would be an account which is free but _can_ be paid for incrementally ("investment"). Money could go to charity, whatever. Servers with which the client authorizes using a stevenarella account could set a minimum account investment. The point is that griefers are less of a problem if you can ban their paid-for-accounts. Minecraft accounts would be considered investment of 20 (?; €/$?), with the ability to link to a stevenarella account once and add that investment to it.

Curious what you think! :)

**Comment by Kezii** [comment]
I think that if you want a drop-in replacement for minecraft, you should probably look to minetest, they have an actual game with working modding and a decent community

BTW, stevenarella already works without microsoft account, making a stevenarella account doesn't really make sense and it's probably illegal

**Reply by PureTryOut** [reply]
Minetest is _not_ a drop-in replacement for Minecraft. Don't get me wrong, I love the project, but it's really not an alternative. The game requires mods to get anything playable and although MineClone does it's very best to implement Minecraft in it, it's just not even close. It feels different, it plays different, it works different.

No, I rather have a fully FOSS Minecraft stack than Minecraft in Minetest.

**Comment by Kezii** [comment]
And yes, contributing makes sense, it's the only source of "minecraft primitives" that we have in rust, I used the protocol library for one of my experiments, and with all its kinks, it worked

If it wasn't for this I had to do everything in java, so thanks iceiix :P

**Comment by ZainlessBrombie** [comment]
Mmh... that one's in C++ and the idea is that both standard clients and stevenarella/offline clients could connect. But I would need a way to prevent spamming griefer accounts...

Why do you think a stevenarella account would be illegal? After all, we are not using anything MS made, right? :)

Huh maybe I'd be better off using feather + this? My plugin uses next to no vanilla features anyways :thinking:

**Comment by iceiix** [comment]
@ZainlessBrombie Thank you for your interest in this project. My opinion, although not the most active, I think it is definitely worth contributing, though I am biased :)

I'm not sure about the Stevenarella account idea, either... having any kind of financial structure introduces many challenges.

But I agree how to make Stevenarella into a sustainable project is a problem worth solving. I'm thinking our best bet is to build enough momentum to attract motivated volunteer developers. Currently it is mostly only me when I find some to work on this project, and occasionally other contributors submitting pull requests. 

"Mobs, graphics, quality of life, chat" are absolutely desirable, certainly could use your help implementing these features/enhancements. Chat especially would be useful to move towards a more playable/interactive state. 

Relevant issues: 
https://github.com/iceiix/stevenarella/issues/63 Chat
https://github.com/iceiix/stevenarella/issues/368 Mobs

---

Also, GitHub introduced this new "Discussions" feature, just enabled it and converted this issue to a discussion thread - welcome to the forums.

**Reply by ZainlessBrombie** [reply]
Oh neat! Yea never used discussions before.

Huh well... I very much do see what you mean by financial structure introducing challenges.

The idea was not to use the money to sustain the stevenarella project - but to just pipe it through to charity or even not handle it in the first place, and accept receipts - the idea is account investment, not financial gain :)

**Reply by ZainlessBrombie** [reply]
Also I hats off for maintaining this neat project :blush:
